File: backend/src/main/java/br/ufpr/sept/so2/modules/academico/api/dto/PeriodoLetivoRequest.java
package br.ufpr.sept.so2.modules.academico.api.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;

public record PeriodoLetivoRequest(
        @NotNull @Min(2000) @Max(2100) Integer ano,
        @NotNull @Min(1) @Max(2) Integer semestre,
        @NotNull LocalDate inicio,
        @NotNull LocalDate fim,
        Boolean ativo
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/academico/api/dto/PeriodoLetivoRequest.kt
package br.ufpr.sept.so2.modules.academico.api.dto

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import java.time.LocalDate

data class PeriodoLetivoRequest(
    @field:NotNull @field:Min(2000) @field:Max(2100) val ano: Int? = null,
    @field:NotNull @field:Min(1) @field:Max(2) val semestre: Int? = null,
    @field:NotNull val inicio: LocalDate? = null,
    @field:NotNull val fim: LocalDate? = null,
    val ativo: Boolean? = null,
)
